### ISSUE TITLE
Kudu can't deploy .NET Core Console apps

### ISSUE THREAD

**Issue by davidmatson** [projectkudu/kudu#2518]
(Moved from comment on #2344)

Kudu doesn't know how to deploy a Console App (.NET Core) project as a web job the way it does for Console App (.NET Framework).

The error message is something like:
`
remote: 'D:\home\site\repository\DotNetCore.csproj' is not a deployable project.
`

I believe the problem lies here:
https://github.com/projectkudu/kudu/blob/af1b4c524926ed0ad9ec6cb5b3789157256eaaf6/Kudu.Core/Infrastructure/VsHelper.cs#L141
This XML namespace isn't used for Common Project System-style csprojs.

(Let me know if any additional data is needed; I could add exact repro steps and exact error output, but I suspect the above data may be sufficient.)

**Comment by watashiSHUN** [comment]
the error message you are seeing is likely because nowadays we only deploy dot net core **web project**, but not console app. I believe it is due to the fact that webjob sdk was not moved to **dotnet core yet** https://stackoverflow.com/questions/43268344/azure-webjob-in-net-core

@davidebbo, but now that https://github.com/Azure/azure-webjobs-sdk/issues/1177 is closed, we can start talking about adding this support?

@davidmatson  can we get you unblocked for now if moving to .net framework or is it a lot of work?

**Comment by davidmatson** [comment]
In case it makes a difference, the particular combination I use in practice is a .NET Core-style csproj but targeting full .NET Framework. To be technically more precise, It's a Common Project System csproj with a target framework of net462 (or similar).

I reverted to old-style csprojs for now with project.json, but that's been deprecated, and I'd like to move to the newer <PackageReference/> csproj syntax if I could.

(Hopefully the work to support .NET Core Console apps proper also covers that style project with a target framework of full .NET, but if not I can split that into a separate issue. I figured saying I'd make this issue just be ".NET Core" support since it's a simpler place to start and likely to get what I needed anyway.)

**Comment by watashiSHUN** [comment]
@davidmatson do you have a github repo that I can use to reproduce the issue? thanks

**Comment by davidmatson** [comment]
Here's a repro for .NET Core console app:
[netcorerepo.zip](https://github.com/projectkudu/kudu/files/1209918/netcorerepo.zip)

**Comment by davidebbo** [comment]
@davidmatson sharing a GitHub repo is highly preferred over a zip file. See https://github.com/projectkudu/kudu/wiki/Using-a-git-repo-to-report-an-issue

**Comment by davidmatson** [comment]
Sure; I'll do that also. (Adding the zip here since I don't plan on keeping the repos after we close this issue.)
[fulldotnetcpsrepro.zip](https://github.com/projectkudu/kudu/files/1209931/fulldotnetcpsrepro.zip)

**Comment by davidmatson** [comment]
Here they are:
https://github.com/davidmatson/kuduissue2518repro1
https://github.com/davidmatson/kuduissue2518repro2

**Comment by watashiSHUN** [comment]
@davidmatson, you initial observation is right on point,
I did a quick hack (add the namespace), unfortunately it didn't work
Instead of getting a `non-deployable project` error, kudu will display this:

> error MSB4041: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 format, please add xmlns="http://schemas.microsoft.com/developer/msbuild/2003" to the <Project> element. If the project has been authored in the old 1.0 or 1.2 format, please convert it to MSBuild 2003 format.

Since msbuild15 is not turned on by default, you need to write a custom deployment, I have an [example based on your repo](https://github.com/watashiSHUN/dotnetcorewebjob)

1. use `dotnet core cli` to build
2. since kudu  does not yet support dotnet core console app (with dlls) I have to output an exe

**Comment by davidmatson** [comment]
Thanks! Yeah, if it wasn't an exe already it should have been (that's what I always use anyway).

Any thoughts on when msbuild15 will be turned on by default? (I'm more likely to wait for it than change back to a custom deploy.cmd - I've tried hard to avoid that in the past.)

**Comment by jvano** [comment]
Hi

If the problem persists and is related to running it on Azure App Service, please open a support incident in Azure:
https://learn.microsoft.com/en-us/azure/azure-portal/supportability/how-to-create-azure-support-request

This way we can better track and assist you on this case

Thanks,

Joaquin Vano
Azure App Service
